include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h

```diff
@@ -724,124 +714,127 @@ __radix_sort_reorder_submit(_ExecutionPolicy&& __exec, ::std::size_t __segments,
-    using _RadixBitsType = ::std::integral_constant<::std::uint32_t, __radix_bits>;
-    using _AscendingType = ::std::bool_constant<__is_comp_asc>;
-    using _CustomName = typename __decay_t<_ExecutionPolicy>::kernel_name;
-    using _RadixCountKernel =
-        oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_generator<__radix_sort_count_kernel, _CustomName,
-                                                                               _RadixBitsType, _AscendingType,
-                                                                               __decay_t<_InRange>, __decay_t<_TmpBuf>>;
-    using _RadixLocalScanKernel =
-        oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_generator<__radix_sort_scan_kernel_1, _CustomName,
-                                                                               _RadixBitsType, __decay_t<_TmpBuf>>;
-    using _RadixGlobalScanKernel =
-        oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_provider<__radix_sort_scan_kernel_2<_CustomName>>;
-    using _RadixReorderPeerKernel = oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_generator<
-        __radix_sort_reorder_peer_kernel, _CustomName, _RadixBitsType, _AscendingType, __decay_t<_InRange>,
-        __decay_t<_OutRange>>;
-    using _RadixReorderKernel = oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_generator<
-        __radix_sort_reorder_kernel, _CustomName, _RadixBitsType, _AscendingType, __decay_t<_InRange>,
-        __decay_t<_OutRange>>;
-
-    ::std::size_t __max_sg_size = oneapi::dpl::__internal::__max_sub_group_size(__exec);
-    ::std::size_t __scan_wg_size = oneapi::dpl::__internal::__max_work_group_size(__exec);
-    ::std::size_t __block_size = __max_sg_size;
-    ::std::size_t __reorder_sg_size = __max_sg_size;
-
-    // correct __block_size, __scan_wg_size, __reorder_sg_size after introspection of the kernels
+template <::std::uint32_t __radix_bits, bool __is_comp_asc, bool __even>
+struct __parallel_radix_sort_iteration {
+
+    template <typename... _Name>
+    using __count_phase = __radix_sort_count_kernel<__radix_bits, __is_comp_asc, __even, _Name...>;
+    template <typename... _Name>
+    using __local_scan_phase = __radix_sort_scan_kernel_1<__radix_bits, _Name...>;
+    template <typename... _Name>
+    using __reorder_peer_phase =  __radix_sort_reorder_peer_kernel<__radix_bits, __is_comp_asc, __even, _Name...>;
+    template <typename... _Name>
+    using __reorder_phase =  __radix_sort_reorder_kernel<__radix_bits, __is_comp_asc, __even, _Name...>;
+
+    template <typename _ExecutionPolicy, typename _InRange, typename _OutRange, typename _TmpBuf>
+    static sycl::event
+    submit(_ExecutionPolicy&& __exec, ::std::size_t __segments, ::std::uint32_t __radix_iter,
+           _InRange&& __in_rng, _OutRange&& __out_rng, _TmpBuf& __tmp_buf, sycl::event __dependency_event)
+    {
+        using namespace oneapi::dpl::__par_backend_hetero::__internal;
```

Comment:
I don't think that `using namespace` is the right change. We already did the investigation with name clashing in the past. I've made the investigation previously and we had an agreement that we would never use `using namespace` in our code ever.
The guide was to introduce the minimal number of namespaces required for the lookup. I believe that if you are already in `oneapi::dpl` you may use just `__par_backend_hetero` and it should be found.

Reply:
> we had an agreement that we would never use using namespace in our code ever.

Thanks for reminding. I will think of a short namespace alias here instead.